docs/coding-guidelines.md

```diff
@@ -354,16 +354,24 @@ Test class names end with `Test` and live in a similar namespace as the class be
 
 #### Unit test method naming
 
-Unit test method names must be descriptive about *what is being tested*, *under what conditions*, and *what the expectations are*. Pascal casing and underscores can be used to improve readability. The following test names are correct:
+Unit test method names must be descriptive about *what is being tested*, *under what conditions*, and *what the expectations are*.
+
+All *new* test cases should follow this convention:
+Pascal casing and underscores should be used to improve readability.
+
+> Format: `<what is being tested>_<under what conditions>_<with what expectations>`
+
+The following test names are correct:
 
 ```cs
-PublicApiArgumentsShouldNotBeNull
-Public_api_arguments_should_not_be_null
+PublicApi_Arguments_ShouldNotBeNull
```

Comment:
Sometimes we test many assert inside 1 unit test method. In that case what would be name?
Can we still do it?

Reply:
I think this is just a random example. The important part is to be descriptive. Obviously for a `[Theory]` you have many "assertions", but describing the pattern in the name is the objective.

Reply:
my take is, the name should describe "what your assertions mean", and not try to indicate all the assertions. "ShouldNotBeNull" sounds like it's 1:1 with `Assert.NotNull`, but I don't think that's our guideline.

Reply:
Fully in agreement with what @donnie-msft said.